src/cnd/cnd.ts

```diff
@@ -44,6 +44,50 @@ export interface SwapRequest {
   peer: Peer;
 }
 
+export interface CoreRequestBody {
+  // to_public_key: string; // This is exchanged via libp2p
+  // secret_hash: string; // Generated by cnd
+  // alpha_ledger: string; // Ledger name is in the route
+  // beta_ledger: string; // Ledger name is in the route
+  alpha_amount: string; // Asset name is in the route
+  beta_amount: string; // Asset name is in the route
+  // final_cltv_delta: number; // beta_expiry
+  // cltv_expiry: number; // beta_expiry
```

Comment:
Two times beta_expiry here?

Reply:
It's because initially I was only building the payload for `HanEthereumEtherHalightLightningBitcoin`.

At the end, this is the `alpha/beta_expiry` so we do not need to include it in the payload. You did notice this is commented right?

Reply:
I was referring to `final_cltv_delta` and `cltv_expiry` both saying "beta_expiry". I assumed this meant that the values of these fields go into "beta_expiry" but we need both for LN so how is this going to work?

Reply:
Ah, ok I think I start to understand the issue.

`final_cltv_delta` and `cltv_expiry` are the same things and must have the same value so if we are doing HanHALight then `beta_expiry` holds the value for both (recip. HALightHAn, `alpha_expiry` holds the value).

Reply:
I am indecisive here. Should we name it `beta_cltv_expiry` then? I feel like there should be a strong link in the name to what you should put here.

Reply:
It's a question of how precise you want the name to be on a given route vs homogeneity across the routes.
I don't know, maybe one of the other reviewers wants to chime in?

Reply:
@bonomat @luckysori @da-kami !

Reply:
I have to admit that I don't understand why you would name it `beta_cltv_expiry` instead of `cltv_expiry` @thomaseizinger . From @D4nte 's comment I would have understood that it could be for both alpha or beta.
I might need more explanation for this to understand the problem so I can give meaningful advice.

Reply:
It would be prefixed with alpha or beta depending in which ledger Lightning is!

That is kind of the combination of:
- use domain specific wording to be clearer
- use alpha/beta to be unambigous

Reply:
OK, I get it. That would mean the `RequestBody` would be different depending on if Lightning is alpha or beta. 

One follow up question:
Is the defined request body only for cases where Lightning is beta ledger? If so I think we should make it specific like suggested by @thomaseizinger

Reply:
It's a currently defined across but we can change.

Reply:
Can someone remind me what expiry we use for Bitcoin and Ethereum? Absolute or relative?

Reply:
I would also advocate for no alpha or beta terminology and just have one interface per protocol combination.

Reply:
> Can someone remind me what expiry we use for Bitcoin and Ethereum? Absolute or relative?

Absolute timestamps.

Reply:
> I would also advocate for no alpha or beta terminology and just have one interface per protocol combination.

I don't think your proposal is feasible. Please provide example on what the fields would be for han/han combination.

Reply:
Please see 60ed7015 for a new proposal.

Reply:
>  Please provide example on what the expiry fields would be for han/han combination

For HAN bitcoin: `bitcoin_expiry`.
For HAN ether: `ethereum_expiry`.

Reply:
> > Please provide example on what the expiry fields would be for han/han combination
> 
> For HAN bitcoin: `bitcoin_expiry`.
> For HAN ether: `ethereum_expiry`.

And for halight?

Reply:
> For HAN bitcoin: bitcoin_expiry.
> For HAN ether: ethereum_expiry.

The downside of this naming-scheme is that it upfront prevents any combinations that act on the same ledger (btc<->ln for example). But maybe that limitation is ok, we probably cannot cater for everything.